vendor/vendor-android/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/TestRunResultsListener.kt

```diff
@@ -44,62 +46,70 @@ class TestRunResultsListener(private val testBatch: TestBatch,
         attachments[test]!!.add(attachment)
     }
 
-    private val logger = MarathonLogging.logger("TestRunResultsListener")
+    override fun getResults(): TestBatchResults {
+        synchronized(runResult) {
```

Comment:
We are trying to write non blocking code using coroutines/actors and etc. If it is possible we need to remove synchronized and use another way